### ISSUE TITLE
BlockGetter using Postgres DB

### ISSUE THREAD

**Issue by InoMurko** [omisego/elixir-omg#1122]
During a POC I noticed that `OMG.Watcher.BlockGetter` is calling `last_persisted_block = DB.Block.get_max_blknum()`. And I wonder why... this is the only occurrence where "security" properties of Watcher are not separated by using PG.

**Comment by InoMurko** [comment]
BlockGetter core provides more insight
```
@doc """
  Ensures the same block will not be send into WatcherDB again.

  Statefull validity keeps track of consumed blocks in separate than WatcherDB database. These databases
  can get out of sync, and then we don't want to send already consumed blocks which could not succeed due
  key constraints on WatcherDB.
  """
```

**Comment by InoMurko** [comment]
OK. Further research shows that BlockGetter is indeed coupled with Postgres.
```
case Core.validate_executions(tx_exec_results, block_application, state) do
      {:ok, state} ->
        block_application
        |> Core.ensure_block_imported_once(state)
        |> Enum.each(&DB.Transaction.update_with/1)

        {:noreply, state, {:continue, {:apply_block_step, :run_block_download_task, block_application}}}

      {{:error, _} = error, new_state} ->
        :ok = update_status(new_state)
        _ = Logger.error("Invalid block #{inspect(block_application.number)}, because of #{inspect(error)}")
        {:noreply, new_state}
    end
```

**Comment by InoMurko** [comment]
Another key point is that block getter tests seem unclean (as in - they don't respect borders between application modes). Block getter is tested by fetching informational API `/account.get_utxos`.

**Comment by pnowosie** [comment]
Yes, indeed BlockGetter is not cleanly separated. `last_persisted_block` seems like security property but in fact is just idempotency / optimization mechanism for Postgres persistence layer.  We don't want to import already imported blocks.

**Comment by InoMurko** [comment]
Can `last_persisted_block` be retrieved from somewhere else? (ignoring postgres)

**Comment by pnowosie** [comment]
> Can `last_persisted_block` be retrieved from somewhere else? (_ignoring postgres_)

Sorry, I might be not precise enough - `last_persisted_block` only tracks the last block available in Postgres DB. It has no meaning to the security critical part which has its own data in RocksDB.
So if you willing to ignore Postgres you can ignore it as well (not store it in process state) or fetch it from Postgres every time its needed.

**Comment by InoMurko** [comment]
That's the thing. Ignoring it is `not trivial` because there's code expecting this field to be populated. The process struct name is `last_block_persisted_from_prev_run`. Question again... can this be swapped by RocksDB `last_block_persisted_from_prev_run`?

**Comment by InoMurko** [comment]
TLDR:
so `last_persisted_block` was only used so that we've ensured the block is not inserted twice into postgres and causing key contraints. with _separation_ of concerns, this is not block getters responsibility and should be pushed on the bus. the consumer should do necessary checks to avoid inserting the block twice.

Thanks @pnowosie ! cc @unnawut
